tabular/src/autogluon/tabular/register/_model_register.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import annotations
+
+from typing import Type
+
+import pandas as pd
+
+from autogluon.core.models import AbstractModel
+
+
+# TODO: Move to core? Maybe TimeSeries can reuse?
```

Comment:
In TimeSeries, we don't have model aliases. The name that needs to be specified by the user is the same as the model class (e.g., `hyperparameters={"DeepAR": {}}` to use the model class `DeepARModel`). I guess it makes sense to have a register to keep track of different available models, but we probably don't need to rely on the same implementation given the different requirements.